gina-alaska/viirs-snow-metrics
# Preprocess snow year

This PR accomplishes a few things:
 - Refreshes to the conda env to use `conda-forge` and adds geospatial packages
 - Modifies top-level config to prescribe a a `SNOW_YEAR` environment variable. A variable value of 2015 would refer to a snow year starting on August 1, 2015 and ending on July 31, 2016. Leap days / years included.
 - Modifies download script to chunk requests on a monthly basis.
 - Adds a preprocess script that combines the downloaded GeoTIFFs into a single netCDF dataset that may then be chunked / queried to compute the actual snow metrics.

To test this PR:
- Read the README :)
- Set your env vars, e.g.,:
 ```sh
export SNOW_YEAR=2015
export OUTPUT_DIR=/$HOME/VIIRS_snow_metrics/outputs
export SCRATCH_DIR=$HOME/VIIRS_snow_metrics/scratch
export INPUT_DIR=/export/datadir/parr_viirs/VIIRS_L3_snow_cover
```
- Rebake your conda env if you already have one, or make a new one from the `environment.yml` file tracked here and activate.
- (optional) run `python download.py` to download the input VIIRS data. This can take several hours. Alternatively, set $INPUT_DIR to `/export/datadir/parr_viirs/VIIRS_L3_snow_cover` or ask me to shoomp pre-downloaded data over to Atlas for your testing leisure.
- Run the preprocessing script (this will take several minutes) with an argument indicating the tile you want to operate on: `python preprocess.py h11v02`
- Browse `preprocessing.log`
- Verify that the file was created where you expect, `ls $SCRATCH_DIR/preprocessed/`

Closes #4
